convert2rhel/checks.py

```diff
@@ -16,13 +16,233 @@
+        assert exit_code == 0
+        kmod_str, exit_code = run_subprocess(
+            'find /lib/modules/{kver} -name "*.ko*"'.format(
+                kver=kernel_version.rstrip("\n")
+            )
+        )
+        assert exit_code == 0
+        assert kmod_str
+    except (subprocess.CalledProcessError, AssertionError):
+        logger.critical("Can't get list of kernel modules.")
+    else:
+        return set(
+            _get_kmod_comparison_key(path)
+            for path in kmod_str.rstrip("\n").split()
+        )
+
+
+def _repos_version_key(pkg_name):
+    try:
+        rpm_version = KERNEL_REPO_RE.search(pkg_name).group("version")
+    except AttributeError:
+        logger.critical(
+            "Unexpected package:\n%s\n is a source of kernel modules.",
+            pkg_name,
+        )
+    else:
+        return tuple(
+            map(
+                _convert_to_int_or_zero,
+                KERNEL_REPO_VER_SPLIT_RE.split(rpm_version),
+            )
+        )
+
+
+def get_most_recent_unique_kernel_pkgs(pkgs):
+    """Return the most recent versions of all kernel packages.
+
+    When we scan kernel modules provided by kernel packages
+    it is expensive to check each kernel pkg. Since each new
+    kernel pkg do not deprecate kernel modules we only select
+    the most recent ones.
+    
+    All RHEL kmods packages starts with kernel* or kmod*
+
+    For example, we have the following packages list:
+        kernel-core-0:4.18.0-240.10.1.el8_3.x86_64
+        kernel-core-0:4.19.0-240.10.1.el8_3.x86_64
+        kmod-debug-core-0:4.18.0-240.10.1.el8_3.x86_64
+        kmod-debug-core-0:4.18.0-245.10.1.el8_3.x86_64
+    ==> (output of this function will be)
+        kernel-core-0:4.19.0-240.10.1.el8_3.x86_64
+        kmod-debug-core-0:4.18.0-245.10.1.el8_3.x86_64
+    
+    _repos_version_key extract the version of a package
+        into the tuple, i.e.
+        kernel-core-0:4.18.0-240.10.1.el8_3.x86_64 ==>
+        (4, 15, 0, 240, 10, 1)
+    
+
+    :type pkgs: Iterable[str]
+    :type pkgs_groups:
+        Iterator[
+            Tuple[
+                package_name_without_version,
+                Iterator[package_name, ...],
+                ...,
+            ]
+        ]
+
+
+    """
+
+    pkgs_groups = itertools.groupby(
+        pkgs, lambda pkg_name: pkg_name.split(":")[0]
+    )
+    return (
+        max(distinct_kernel_pkgs[1], key=_repos_version_key)
+        for distinct_kernel_pkgs in pkgs_groups
+        if distinct_kernel_pkgs[0].startswith(("kernel", "kmod"))
+    )
+
+
+def get_rhel_supported_kmods():
+    """Return set of target RHEL supported kernel modules."""
+    repoquery_repoids_args = " ".join(
+        "--repoid " + repoid for repoid in system_info.get_enabled_rhel_repos()
+    )
+    # get output of a command to get all packages which are the source
+    # of kmods
+    kmod_pkgs_str, _ = run_subprocess(
+        (
+            "repoquery "
```

Comment:
You need to handle the repoquery commands in a special way on CentOS/OL 8, otherwise you'll be getting:
```
Modular dependency problems:

 Problem 1: conflicting requests
  - nothing provides module(platform) needed by module ant:1.10:820181213135032:5ea3b708-0.x86_64
 Problem 2: conflicting requests
  - nothing provides module(platform:el8) needed by module container-tools:rhel8:8030020201124131330:830d479e-0.x86_64
```

For yum calls, we use the `--setopt=module_platform_id=platform:el8` option: https://github.com/oamg/convert2rhel/blob/main/convert2rhel/pkghandler.py#L113.